src/System.Composition.Hosting/src/System/Composition/Hosting/Core/ExportDescriptorRegistryUpdate.cs

```diff
@@ -128,8 +129,10 @@ private StringBuilder DescribeCompositionStack(CompositionDependency import, IEn
 
         protected override IEnumerable<ExportDescriptorPromise> GetPromises(CompositionContract contract)
         {
-            Assumes.IsTrue(!_updateFinished, "Update is finished - dependencies should have been requested earlier.");
-
+            if (_updateFinished)
+            {
+                throw new Exception(String.Format(CultureInfo.CurrentCulture, SR.Dependencies_Should_Be_Requested_Earlier));
```

Comment:
you don't need the string.format here as there is no culture formatting involved: `throw new Exception(SR.Dependencies_Should_Be_Requested_Earlier);`